Core/include/Acts/Utilities/HashedString.hpp

```diff
@@ -0,0 +1,52 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <string_view>
+#include <utility>
+
+namespace Acts {
+using HashedString = std::uint32_t;
+
+// Adapted from https://gist.github.com/Lee-R/3839813
+namespace detail {
+// FNV-1a 32bit hashing algorithm.
+constexpr HashedString fnv1a_32(char const* s, std::size_t count) {
```

Comment:
Good point. I'm not sure. How would you suggest to do such a detection?

Reply:
This would require an extra runtime check every time a component is accessed.

Looking at [this](https://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed/145633#145633) the author found four collisions out of about 200k english words. I think at leading order we should probably be ok without an extra check?

What do you think?

Reply:
:-) - given we use ~common words and probably only 10 of them .. each. Let's forget about it. I think we would capture it in unit testing anyways.

Reply:
Agreed. Resolving this.